include/antler/project/project.hpp

```diff
@@ -49,132 +63,162 @@ class project {
    /// @param ver  The new version information.
    void version(const antler::project::version& ver) noexcept;
 
-   /// Remove a named object of a given type. If type is any, then all objects with name are removed.
-   /// @param name  The name of the object(s) to remove.
-   /// @param type  The type of the object(s) to remove. If type is any, then all objects with name are removed.
-   /// @return True if one or more objects are removed; otherwise, false.
-   [[nodiscard]] bool remove(std::string_view name, object::type_t type) noexcept;
+   /// Remove an app from the project.
+   /// @param name  The name of the app to remove.
+   /// @return True if the app existed and was erased, false otherwise.
+   [[nodiscard]] inline bool remove_app(const std::string& name) noexcept { return m_apps.erase(name) > 0; }
+
+   /// Remove a lib from the project.
+   /// @param name  The name of the lib to remove.
+   /// @return True if the lib existed and was erased, false otherwise.
+   [[nodiscard]] inline bool remove_lib(const std::string& name) noexcept { return m_libs.erase(name) > 0; }
+
+   /// Remove an object from the project.
+   /// @param name  The name of the object to remove.
+   /// @return True if the object existed and was erased, false otherwise.
+   [[nodiscard]] inline bool remove(const std::string& name) noexcept { return remove_app(name) || remove_lib(name); }
 
    /// update or insert a new object. It's type() is evaluated to determine which list it belongs in. If the object already
-   /// exists, an update is performed by removing the old one and adding the new one.
+   /// exists.
    /// @param obj  The object to update or insert.
-   template <object::type_t Ty>
-   inline void upsert(object&& obj) noexcept {
-      if constexpr (Ty == object::type_t::app)
+   template <typename Tag>
+   inline void upsert(object<Tag>&& obj) noexcept {
+      if constexpr (std::is_same_v<Tag, app_tag>)
          return upsert_app(std::move(obj));
-      else if constexpr (Ty == object::type_t::lib)
+      else if constexpr (std::is_same_v<Tag, lib_tag>)
          return upsert_lib(std::move(obj));
-      else if constexpr (Ty == object::type_t::test)
-         return upsert_test(std::move(obj));
+      //else if constexpr (Ty == object::type_t::test)
+      //   return upsert_test(std::move(obj));
```

Comment:
any reason for keeping commented out code?

Reply:
Added comment in 866aa28